### PR TITLE
Add Shrinkage Loss to handle regression label imbalance problem.

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*

*Description of changes:*
This PR follows the paper: https://openaccess.thecvf.com/content_ECCV_2018/html/Xiankai_Lu_Deep_Regression_Tracking_ECCV_2018_paper.html to implement the Shrinkage Loss for imbalanced regression tasks.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/config/argument.py [hunk 1]
```diff
@@ -2871,6 +2875,21 @@ def class_loss_func(self):
...
+    @property
+    def regression_loss_func(self):
+        """ Regression loss function. Builtin loss functions include
+            ``mse`` and ``shrinkage``. Default is ``mse``.
+        """
+        # pylint: disable=no-member
+        if hasattr(self, "_regression_loss_func"):
+            assert self._regression_loss_func in BUILTIN_REGRESSION_LOSS_FUNCTION, \
+                f"Only support {BUILTIN_REGRESSION_LOSS_FUNCTION} " \
+                "loss functions for classification tasks"
```
Comment (jalencato) [inline python/graphstorm/config/argument.py:2887]: regression tasks?

File: python/graphstorm/model/loss_func.py [hunk 2]
```diff
@@ -222,6 +222,93 @@ def out_dims(self):
...
+        For more details, please refer to the paper
+        "Deep Regression Tracking with Shrinkage Loss"
+        (https://openaccess.thecvf.com/content_ECCV_2018/html/Xiankai_Lu_Deep_Regression_Tracking_ECCV_2018_paper.html)
+
+        Parameters
+        ----------
+        alpha: float
+            A hyper-parameter controls the loss shrinkage
+            speed when the prediction error decreases.
+            Default: 10.
```
Comment (jalencato) [inline python/graphstorm/model/loss_func.py:253]: ``10``, add `` here as well.

File: python/graphstorm/model/loss_func.py [hunk 3]
```diff
@@ -222,6 +222,93 @@ def out_dims(self):
...
+            The prediction results.
+        labels: torch.Tensor
+            The training labels.
+
+        Returns
+        -------
+        loss: Tensor
+            The loss value.
+        """
+        # Make sure the lable is a float tensor
```
Comment (jalencato) [inline python/graphstorm/model/loss_func.py:282]:
```suggestion
        # Make sure the lables is a float tensor
```

File: tests/unit-tests/test_loss.py [hunk 4]
```diff
@@ -108,8 +109,29 @@ def test_FocalLossFunc():
     assert_almost_equal(loss.numpy(), gt_loss.numpy(), decimal=4)
 
 
+def test_ShrinkageLossFunc():
+    alpha = 10.
+    gamma = 0.2
+    loss_func = ShrinkageLossFunc(alpha, gamma)
+    logits = th.tensor([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0])
+    labels = th.tensor([1., 1., 1., 1., 1., 1., 1., 1., 0, 0.9])
+    gt_loss = th.tensor(0.3565)
```
Comment (jalencato) [inline tests/unit-tests/test_loss.py:118]: We can paste the loss = \frac{l^2}{1 + \exp \left( \alpha \cdot (\gamma - l) \right)} here, about how we get the value.

File: python/graphstorm/config/argument.py (hunk 1 above)
Comment (classicsong) [reply python/graphstorm/config/argument.py:2887]: yes

**Review by jalencato — APPROVED** [review]
LGTM.

File: python/graphstorm/gsf.py [hunk 5]
```diff
@@ -521,7 +524,17 @@ def create_builtin_node_decoder(g, decoder_input_dim, config, train_task):
...
+            loss_func = RegressionLossFunc()
+        elif config.regression_loss_func == BUILTIN_REGRESSION_LOSS_SHRINKAGE:
+            # set default value of alpha to 10. for shrinkage loss
+            # set default value of gamma to 0.2 for shrinkage loss
+            alpha = config.alpha if config.alpha is not None else 10.
+            gamma = config.gamma if config.gamma is not None else 0.2
+            loss_func = ShrinkageLossFunc(alpha, gamma)
+        else:
+            raise RuntimeError(
+                        f"Unknown regression loss {config.class_loss_func}")
```
Comment (zhjwy9343) [inline python/graphstorm/gsf.py:537]:
```suggestion
                        f"Unknown regression loss {config.regression_loss_func}")
```

File: python/graphstorm/gsf.py [hunk 6]
```diff
@@ -723,7 +736,18 @@ def create_builtin_edge_decoder(g, decoder_input_dim, config, train_task):
...
```
Comment (zhjwy9343) [inline python/graphstorm/gsf.py:750]:
```suggestion
                        f"Unknown regression loss {config.regression_loss_func}")
```

**Review by zhjwy9343 — APPROVED** [review]
LGTM
